### ISSUE TITLE
Code reviewing: examples of good practice

### ISSUE THREAD

**Issue by sofer** [foundersandcoders/master-reference#90]
I was just discussing this with @eliascodes. We do already have [Notes on code reviewing](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/general/code-reviews.md), but it would be useful to see examples of good practice in actual pull requests. Has anyone got any suggestions for their favourite ever discussion in a pull request on an open source repo? 

Please post your links here @iteles @nelsonic @besarthoxhaj @izaakrogan @FilWisher @NataliaLKB @benjaminlees @rjmk @nikhilaravi @jrans @Danwhy @minaorangina @heron2014 @SimonLab @des-des @jackcarlisle @sohilpandya @naazy @Conorc1000 @mantagen @Jbarget and all...

**Comment by jackcarlisle** [comment]
Here's an example of a UX/UI feature PR that stood out from my time at Thomas Cook. The number of options within the screenshots made it easy to discuss and in turn review. I'm not sure if it's best practice to keep all of this in the Pull Request itself or to keep a record of it in the issue that the PR relates to but it was a good example of how you can make the review clear and easy for the person reviewing it: 

https://github.com/numo-labs/isearch-ui/pull/467

**Comment by jsms90** [comment]
See [my advice](https://github.com/foundersandcoders/master-reference/pull/409#issuecomment-312446235) in a `master-reference` PR

> It is best to think of a PR, like everything else on github, as a communication tool between you and your teammates. See [this example of an excellent PR](https://github.com/alphagov/frontend/pull/784)
>
> These articles may help:
> https://www.annashipman.co.uk/jfdi/good-pull-requests.html
> https://gist.github.com/mikepea/863f63d6e37281e329f8
> https://www.atlassian.com/blog/git/written-unwritten-guide-pull-requests

**Comment by jsms90** [comment]
From my interactions with the mentors, I have discovered that many people were not aware of the "review changes" button on a Pull Request
![image](https://user-images.githubusercontent.com/15656538/29406431-f5106cb6-8338-11e7-97d6-8ccfbfc0583f.png)

They didn't know about how to review a PR at all e.g. approve vs comment vs request changes. They didn't know about the ability to make comments against particular lines.

This is partly a reflection of the fact that, during the course, students often explain their PRs to each other in person. At other times, people just take a cursory glance over each other's work & merge because of time pressure.

Ideally, no one would be graduating the course without knowing about reviewing PRs.

**Comment by zurda** [comment]
@jsms90 , I agree. Is it possible to introduce this clarification/ task in the [git workflow workshop for two](https://github.com/foundersandcoders/git-workflow-workshop-for-two/issues/52)? This could be added as either a bonus, or as an addition to step 11 asking the student to request changes in the PR: 

![Screenshot 2019-04-10 at 19 52 38](https://user-images.githubusercontent.com/16784959/55905778-a027e000-5bca-11e9-90f0-24b13b1bb4b0.png)
